src/hooks.ts

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { ErrorContext } from './contexts';
+import { getTypeFromError } from './utils';
+import * as Sentry from '@sentry/react';
+
+// Convenience hook for manually displaying an app-level error screen and notifying Sentry.
+// Takes a plain error object, or null to clear the error.
+export const useSetAppError = () => {
+  const { setError } = React.useContext(ErrorContext);
+
+  return React.useCallback((error: Error | null) => {
+    if (error) {
+      Sentry.captureException(error);
+    }
+    setError(
+      // ErrorBoundary expects a wrapped SentryError
+      error ? {
+        error: error,
+        type: getTypeFromError(error),
+      } : null
```

Comment:
how does the sentry event id get into the state so that it shows up in the dialog?

Reply:
If it's not in the context, the Error component checks for `Sentry.lastEventId` or polls until it's available... I didn't realize `captureException` returns an event ID though, so I can just set that in the context here.